tracing-appender/src/rolling.rs

```diff
@@ -602,7 +602,19 @@ impl Inner {
                     return None;
                 }
 
-                let created = metadata.created().ok()?;
+                let created = metadata.created().ok().or_else(|| {
+                    let mut datetime = filename;
+                    if let Some(prefix) = &self.log_filename_prefix {
+                        datetime = datetime.strip_prefix(&format!("{prefix}."))?;
```

Comment:
This might be an unnecessary optimisation or one that the compiler does anyway but would it make sense to split these strip methods to save building two new strings for every file in a potentially full directory?

```suggestion
                        datetime = datetime.strip_prefix(prefix)?;
                        datetime = datetime.strip_prefix('.')?;
```

Reply:
nope, doesn't get optimized away, LGTM!